### DISCUSSION TITLE
Can i set a baseUrl for defaultLocale

### DISCUSSION THREAD

**Discussion by laojun** [facebook/docusaurus#8745] [Q&A] [answered]
Hi, I have one case here：

Here is my config for i18n
```
defaultLocale: 'zh',
locales: ['en', 'zh'],
```
And i chosen the [single domain](https://docusaurus.io/docs/i18n/tutorial#single-domain-deployment) to deploy my website. But i don't want the default url 'https://my-site/' , Is it possible to visit my website like:
```
https://my-site/en
https://my-site/zh
```

**Comment by slorber** [comment, marked as answer]
Hi,

I understand that you don't want the baseUrl to be removed for the default locale.

For now, we don't make it very easy, unfortunately.

But a new `DOCUSAURUS_CURRENT_LOCALE` env variable available in the config file should make it easy to tweak the config on a per-locale basis (ie use `baseUrl: `/zh/` for locale `zh`). 

Should be available in the upcoming v2.4, but you can try using a canary version today.
https://github.com/facebook/docusaurus/pull/8677


We also want to add first-class support for a baseUrl that you can configure per locale, as it was attempted in this draft PR: https://github.com/facebook/docusaurus/pull/6731

**Reply by laojun** [reply]
It would be awesome!
thanks for the reply.

**Reply by laojun** [reply]
For now, I did it this way, kinda hack

1、set a `default` locale
```
defaultLocale: 'default',
locales: ['en', 'zh', 'default'],
```

2、redirect default url to zh
```
onRouteUpdate({ location }) {
    if (!(location.pathname.startsWith('/zh') || location.pathname.startsWith('/en'))) {
      window.location.href = `/zh${location.pathname}`;
    }
  },
```

3、Swizzling `NavbarItem/LocaleDropdownNavbarItem`, and than filter the `default` item
```
const items = [...dropdownItemsBefore, ...localeItems, ...dropdownItemsAfter].filter(
    (item) => item.lang !== 'default'
  );
```

**Reply by slorber** [reply]
Good workaround until the env variable lands.

You could also avoid building for the default locale

```diff
- yarn build
+ yarn build --locale en
+ yarn build --locale zh
```
